data-node/src/main/java/org/graylog/datanode/bootstrap/preflight/DataNodePreflightGeneratePeriodical.java

```diff
@@ -113,19 +114,20 @@ public void doRun() {
                         KeyStore nodeKeystore = certificateAndPrivateKeyMerger.merge(
                                 certificateChain.get(),
                                 privateKeyEncryptedStorage,
-                                DEFAULT_PASSWORD.toCharArray(),
-                                configuration.getDatanodeHttpCertificatePassword().toCharArray(),
+                                passwordSecret.toCharArray(),
+                                passwordSecret.toCharArray(),
                                 CertConstants.DATANODE_KEY_ALIAS
                         );
 
+                        final var password = configuration.getDatanodeHttpCertificatePassword() != null ? configuration.getDatanodeHttpCertificatePassword().toCharArray() : passwordSecret.toCharArray();
```

Comment:
I have a doubt about this line.
`nodeKeystore` is created with `passwordSecret`.
Should not `password` be equal to `passwordSecret` no matter what now?
Does it work correctly on your side because you have not set `DatanodeHttpCertificatePassword`, or is it me who does not understand it correctly.

Reply:
I believe I have fixed that.